presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerdeFactory.java

```diff
@@ -34,10 +37,27 @@ public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compress
 
     public PagesSerde createPagesSerde()
     {
+        return createPagesSerdeInternal(Optional.empty());
+    }
+
+    public PagesSerde createPagesSerdeForSpill(Optional<SpillCipher> spillCipher)
+    {
+        return createPagesSerdeInternal(spillCipher);
+    }
+
+    private PagesSerde createPagesSerdeInternal(Optional<SpillCipher> spillCipher)
+    {
+        Optional<Compressor> compressor;
+        Optional<Decompressor> decompressor;
         if (compressionEnabled) {
-            return new PagesSerde(blockEncodingSerde, Optional.of(new Lz4Compressor()), Optional.of(new Lz4Decompressor()));
+            compressor = Optional.of(new Lz4Compressor());
+            decompressor = Optional.of(new Lz4Decompressor());
+        }
+        else {
+            compressor = Optional.empty();
+            decompressor = Optional.empty();
         }
 
-        return new PagesSerde(blockEncodingSerde, Optional.empty(), Optional.empty());
+        return new PagesSerde(blockEncodingSerde, compressor, decompressor, spillCipher);
```

Comment:
The original code style in this method is better:

```
    private PagesSerde createPagesSerdeInternal(Optional<SpillCipher> spillCipher)
    {
        if (compressionEnabled) {
            return new PagesSerde(blockEncodingSerde, Optional.of(new Lz4Compressor()), Optional.of(new Lz4Decompressor()), spillCipher);
        }

        return new PagesSerde(blockEncodingSerde, Optional.empty(), Optional.empty(), spillCipher);
    }
```

Reply:
Sure, will change to this implementation.